overlord/snapstate/snapstatetest/devicectx.go

```diff
@@ -119,8 +119,12 @@ func MockDeviceContext(deviceCtx snapstate.DeviceContext) (restore func()) {
 	r1 := ReplaceDeviceCtxHook(deviceCtxHook)
 	// for convenience reflect from the context whether there is a
 	// remodeling
-	r2 := ReplaceRemodelingHook(func(*state.State) bool {
-		return deviceCtx != nil && deviceCtx.ForRemodeling()
+	r2 := ReplaceRemodelingHook(func(*state.State) *state.Change {
+		if deviceCtx != nil && deviceCtx.ForRemodeling() {
+			return &state.Change{}
```

Comment:
Nitpick, this looks slightly odd at first, maybe add a comment or change it to return st.NewChange("dummy", "dummy remodeling change affecting conflict checks") or something like this?

Reply:
Thanks for the feedback. I didn't do it at first to avoid adding side-effects that could break tests but I change it to use state.NewChange and it works fine